tests/integration/cloudformation/test_template_engine.py

```diff
@@ -245,6 +245,23 @@ def test_get_azs_function(self, deploy_cfn_template):
         zone = "us-east-1a"  # TODO parametrize
         assert zone in deployed.outputs["Zones"]
 
+    @pytest.mark.aws_validated
+    @pytest.mark.parametrize("create_parameter", ("true", "false"), ids=("create", "no-create"))
+    def test_conditional_att_to_conditional_resources(self, deploy_cfn_template, create_parameter):
+        template_path = os.path.join(
+            os.path.dirname(__file__), "../templates/cfn_if_attribute_none.yml"
+        )
+
+        deployed = deploy_cfn_template(
+            template_path=template_path,
+            parameters={"CreateParameter": create_parameter},
+        )
+
+        if create_parameter == "false":
+            assert deployed.outputs["Result"] == "Value1"
+        else:
+            assert deployed.outputs["Result"] == "Value2"
```

Comment:
Suggestion: Maybe snapshot this result, making this test a little simpler?
```suggestion
        snapshot.match("output-value", deployed.outputs["Result"])
```